### PR TITLE
Feature/decorate class

### PR DESCRIPTION
**Pull request by daudi13**
In this PR I:

- Created a class Nameable
- Turned my Person class to Nameable
- Prepared the base decorator
- prepared class CapitalizeDecorator 
- prepared TrimmerDecorator

### CODE REVIEW

File: base_decorator.rb [hunk 1]
```diff
@@ -0,0 +1,12 @@
+require './nameable'
+
+class Decorator < Nameable
+  def initialize(nameable)
+    super()
+    @nameable = nameable
+  end
+
+  def correct_name
+    @nameable
```
Comment (t-yanick) [inline base_decorator.rb]: - [x] The correct_name method  should return the result of the correct_name method of  @namable, like so:
 ```ruby
    def correct_name
      @nameable.correct_name
    end
```
Please kindly fix that.

**Review by t-yanick — CHANGES_REQUESTED** [review]
Hi @daudi13, 

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: base_decorator.rb (hunk 1 above)
Comment (daudi13) [reply base_decorator.rb]: thanks for pointing that out.

**Review by Lameck1 — APPROVED** [review]
Hi, @daudi13,

You've done well addressing concerns from the previous review 👏🏼.
Tasks for this project are now successfully implemented 💯.

### Highlights
- Professional README ✔️
- Clean code ✔️ 
- Good commit messages ✔️ 

All the best with the next project.

You are free to merge this branch :shipit:
Congratulations! 🎉 

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
**Please, remember to tag me in your question so I can receive the notification.**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**